packages/core/src/main/mls/MLSService/MLSService.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Wire
+ * Copyright (C) 2022 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {
+  AddProposalArgs,
+  CommitBundle,
+  ConversationConfiguration,
+  ConversationId,
+  CoreCrypto,
+  DecryptedMessage,
+  ExternalProposalArgs,
+  ExternalProposalType,
+  ExternalRemoveProposalArgs,
+  Invitee,
+  ProposalArgs,
+  ProposalType,
+  RemoveProposalArgs,
+} from '@otak/core-crypto/platforms/web/corecrypto';
+import {APIClient} from '@wireapp/api-client';
+import {QualifiedUsers} from '../../conversation';
+import {Decoder, Encoder} from 'bazinga64';
+import type {MLSConfig} from '../types';
+
+//@todo: this function is temporary, we wait for the update from core-crypto side
+//they are returning regular array instead of Uint8Array for commit and welcome messages
+export const optionalToUint8Array = (array: Uint8Array | []): Uint8Array => {
+  return Array.isArray(array) ? Uint8Array.from(array) : array;
+};
+
+export class MLSService {
```

Comment:
The reason I've added MLSService is I wanted to have only one place that has access to coreCryptoClient which is possibly undefined (proteus protocol), so using client through this service would be safe (otherwise error is thrown). This would be good place to store some mls logic shared between different services (`ConversationService` and `NotificationService` for now). Let me know what you think. More context in conversation with @thisisamir98:

<img width="724" alt="Screenshot 2022-09-07 at 12 44 04" src="https://user-images.githubusercontent.com/45733298/188859600-4a7c9f1b-03b6-40e9-8eae-9924aa5c45ed.png">

Reply:
Thanks for the explanation, and I think it's a solid reason to do so. Very good call 👌